skbase/tests/test_base.py

```diff
@@ -170,6 +271,79 @@ def test_set_tags():
     assert FIXTURE_OBJECT_SET.get_tags() == FIXTURE_OBJECT_SET_TAGS, msg
 
 
+def test_set_tags_works_with_missing_tags_dynamic_attribute():
```

Comment:
I would say, it should raise an error instead, because it means the object has not been correctly initialized!

Reply:
I think way back, when we added it to sktime, I had the logic setup to check for this and raise an error and you or another reviewer suggested that the code just initialize the missing attribute.

I don't have strong feelings either way, but this tests a feature of the current set_tags. I'd propose we leave the test, and if we decide to update how set_tags work, we amend the test (remove this test and replace with test that correct error is raised).